joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/AstForPrimitivesCreator.scala

```diff
@@ -35,10 +34,16 @@ trait AstForPrimitivesCreator { this: AstCreator =>
     Ast(createLiteralNode(code, Some(Defines.STRING), stringLiteral.lineNumber, stringLiteral.columnNumber))
   }
 
-  protected def astForSpreadElement(spreadElement: BabelNodeInfo): Ast = {
-    val ast = astForNode(spreadElement.json("argument"))
-    ast.nodes.collectFirst { case i: NewIdentifier => i }.foreach(_.code = spreadElement.code)
-    ast
+  protected def astForSpreadOrRestElement(spreadElement: BabelNodeInfo, arg1Ast: Option[Ast] = None): Ast = {
+    val ast = astForNodeWithFunctionReference(spreadElement.json("argument"))
+    val callNode = createCallNode(
+      spreadElement.code,
+      "<operator>.starredUnpack",
+      DispatchTypes.STATIC_DISPATCH,
+      spreadElement.lineNumber,
+      spreadElement.columnNumber
+    )
+    callAst(callNode, arg1Ast.toList :+ ast)
```

Comment:
I'm not sure when this would even happen, but if this is called with `None` for `arg1Ast`, will the argument indices still be correct? (i.e. will the object that's being spread still have argument index 2?)

Reply:
`.toList` will result in an empty List for `None` so this should work.

Reply:
yes, it won't cause an error. My question is if the argument indices would change in that case. Because that would be a problem when writing policies.

Reply:
No, it would not. Its an empty List not a List(Ast()).